crates/polars-plan/src/logical_plan/mod.rs

```diff
@@ -80,6 +81,7 @@ pub enum DslPlan {
         paths: Arc<[PathBuf]>,
         // Option as this is mostly materialized on the IR phase.
         file_info: Option<FileInfo>,
+        hive_parts: Option<Vec<Arc<HivePartitions>>>,
```

Comment:
Moved this out from `file_info` - we store hive parts for every path in a `Vec` that we can resolve up-front - this replaces the current approach of using `update_hive_partitions`.